File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.cpp
class Solution {
public:
    int beautySum(string s) {
        int res = 0;
        for (int i = 0; i < (int)s.size(); ++i) {
            unordered_map<char,int> freq;
            for (int j = i; j < (int)s.size(); ++j) {
                freq[s[j]]++;

                int ma = 0, mi = INT_MAX;
                for (const auto& kv : freq) {
                    ma = max(ma, kv.second);
                    mi = min(mi, kv.second);
                }
                res += (ma - mi);
            }
        }
        return res;
    }
};

File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.py
class Solution:
    def beautySum(self, s: str) -> int:
        res=0
        for i in range(len(s)):
            freq ={}
            for j in range(i,len(s)):
                if(s[j] in freq):
                    freq[s[j]]+=1
                else:
                    freq[s[j]]=1
                val = freq.values()
                mi = min(val)
                ma = max(val)
                res+=(ma-mi)
        return res